src/Surfnet/StepupGateway/SamlStepupProviderBundle/Controller/SamlProxyController.php

```diff
@@ -295,9 +295,6 @@ public function renderSamlResponse($view, StateHandler $stateHandler, SAMLRespon
             'relayState' => $stateHandler->getRelayState()
         ]);
 
-        // clear the state so we can call again :)
-        $stateHandler->clear();
```

Comment:
Part of the state is the ID from the SAML AuthnRequest. This is used for replay protection. The ID from the AuthnRequest must match the InResponseTo in the Response. It is correct to invalidate the ID after receiving a SAML Response. So resending a SAML Response could result in a different error.

I agree that from a user and support perspective it is nice if reloading an error page does not change the error. But this behaviour can be achieved by other means (e.g. by redirecting to an error URL.

Reply:
Thanks, the proper solution is a but more complicated. I'm closing (and not merging) this PR and created a pivotal issue for further discussion on how to handle this in the future: https://www.pivotaltracker.com/story/show/155850805